lib/storage/src/chunks/getChunkBuffer.ts

```diff
@@ -0,0 +1,22 @@
+import { RawDataPart } from "../Upload";
+
+export async function* getChunk(data: Buffer, partSize: number): AsyncGenerator<RawDataPart> {
```

Comment:
Updates:
* value `void` for `TReturn`: The generator function falls off the end without a return and the caller gets back `{ value: undefined, done: true }`.
* value `undefined` for `TNext`: next function does not accept any parameters.

```suggestion
export async function* getChunk(data: Buffer, partSize: number): AsyncGenerator<RawDataPart, void, undefined> {
```

Reply:
I'll add this for verboseness, but I dont think this is strictly needed right? These types are implied. Are you getting type errors?

Reply:
They're implied to `any` and `unknown` which may cause issues in future https://github.com/microsoft/TypeScript/blob/2428ade1a91248e847f3e1561e31a9426650efee/src/lib/es2018.asyncgenerator.d.ts#L3

I'm not getting any type errors right now.

Reply:
okay gotcha. These are updated